include/vinecopulib/bicop/abstract.hpp

```diff
@@ -47,52 +49,60 @@ class AbstractBicop
 
   virtual void set_parameters(const Eigen::MatrixXd& parameters) = 0;
 
-  virtual void flip() = 0;
-
   // Virtual methods
-  virtual void fit(const Eigen::Matrix<double, Eigen::Dynamic, 2>& data,
+  virtual void fit(const Eigen::MatrixXd& data,
                    std::string method,
                    double mult,
                    const Eigen::VectorXd& weights) = 0;
-
+  
   virtual double get_npars() = 0;
 
   virtual double parameters_to_tau(const Eigen::MatrixXd& parameters) = 0;
 
-  Eigen::VectorXd pdf(const Eigen::Matrix<double, Eigen::Dynamic, 2>& u);
+  virtual void flip() = 0;
 
-  double loglik(const Eigen::Matrix<double, Eigen::Dynamic, 2>& u,
-                const Eigen::VectorXd weights = Eigen::VectorXd());
+  // following are virtual so they can be overriden by KernelBicop
+  virtual Eigen::VectorXd pdf(const Eigen::MatrixXd& u);
```

Comment:
Can you explain why you changed some by adding the ` = 0` and for others you removed it?